tests/catch/unit/memory/hipArray3DCreate.cc

```diff
@@ -0,0 +1,314 @@
+
+TEMPLATE_TEST_CASE("Unit_hipArray3DCreate_happy", "", char, uchar2, uint2, int4, short4, float,
+                   float2, float4) {
+  using vec_info = vector_info<TestType>;
+  DriverContext ctx;
+
+  HIP_ARRAY3D_DESCRIPTOR desc{};
+  desc.Format = vec_info::format;
+  desc.NumChannels = vec_info::size;
+#if HT_AMD
+  desc.Flags = 0;
+#else
+  desc.Flags = GENERATE(0, CUDA_ARRAY3D_SURFACE_LDST);
+#endif
+
+  constexpr size_t size = 64;
+
+  std::vector<hipExtent> extents{
+      {size, 0, 0},       // 1D array
+      {size, size, 0},    // 2D array
+      {size, size, size}  // 3D array
+  };
+
+  for (auto& extent : extents) {
+    desc.Width = extent.width;
+    desc.Height = extent.height;
+    desc.Depth = extent.depth;
+
+    CAPTURE(desc.Width, desc.Height, desc.Depth);
+
+    hiparray array;
+    HIP_CHECK(hipArray3DCreate(&array, &desc));
+    checkArrayIsExpected(array, desc);
+    HIP_CHECK(hipArrayDestroy(array));
+  }
+}
+
+TEMPLATE_TEST_CASE("Unit_hipArray3DCreate_MaxTexture", "", int, uint4, short, ushort2,
+                   unsigned char, float, float4) {
+#if HT_AMD
+  HipTest::HIP_SKIP_TEST("EXSWCPHIPT-97");
+  return;
+#endif
+
+  using vec_info = vector_info<TestType>;
+  DriverContext ctx;
+
+  hiparray array;
+  HIP_ARRAY3D_DESCRIPTOR desc{};
+  desc.Format = vec_info::format;
+  desc.NumChannels = vec_info::size;
+#if HT_AMD
+  desc.Flags = 0;
+#else
+  desc.Flags = GENERATE(0, CUDA_ARRAY3D_SURFACE_LDST);
+  if (desc.Flags == CUDA_ARRAY3D_SURFACE_LDST) {
+    HipTest::HIP_SKIP_TEST("EXSWCPHIPT-58");
+    return;
+  }
+#endif
+  CAPTURE(desc.Flags);
+
+  const Sizes sizes(desc.Flags);
+  CAPTURE(sizes.max1D, sizes.max2D, sizes.max3D);
+
+  const size_t s = 64;
+  SECTION("Happy") {
+    // stored in a vector so some values can be ifdef'd out
+    std::vector<hipExtent> extentsToTest{
+        make_hipExtent(sizes.max1D, 0, 0),                              // 1D max
+        make_hipExtent(sizes.max2D[0], s, 0),                           // 2D max width
+        make_hipExtent(s, sizes.max2D[1], 0),                           // 2D max height
+        make_hipExtent(sizes.max2D[0], sizes.max2D[1], 0),              // 2D max
+        make_hipExtent(sizes.max3D[0], s, s),                           // 3D max width
+        make_hipExtent(s, sizes.max3D[1], s),                           // 3D max height
+        make_hipExtent(s, s, sizes.max3D[2]),                           // 3D max depth
+        make_hipExtent(s, sizes.max3D[1], sizes.max3D[2]),              // 3D max height and depth
+        make_hipExtent(sizes.max3D[0], s, sizes.max3D[2]),              // 3D max width and depth
+        make_hipExtent(sizes.max3D[0], sizes.max3D[1], s),              // 3D max width and height
+        make_hipExtent(sizes.max3D[0], sizes.max3D[1], sizes.max3D[2])  // 3D max
+    };
+    const auto extent =
+        GENERATE_COPY(from_range(std::begin(extentsToTest), std::end(extentsToTest)));
+
+    desc.Width = extent.width;
```

Comment:
Again, suggestion ```desc.Width = GENERATE(s, sizes.max1D, sizes.max2D[0], sizes.max3D[0]); ``` (for desc.Heigh & desc.Depth as well)
I think it improves readability, and in since it is one line(ish) you can always ifdef it with another version of desc.Width.
If you do not like this approach, I'm happy with it's current state as well.

Reply:
I think this would lead to situations with `Width = sizes.max1D` and `Height = sizes.max2D[1]`, which I don't want.